### PR TITLE
[13.0][stock_request_stage][add] new module to manage stock requests using stages

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This module allows you to manage the completion of stock requests in stages.
A Kanban view is added to stock requests, allowing you to move stock requests
across the defined stages.

image

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
@ForgeFlow @HviorForgeFlow

File: stock_request_stage/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2018 Creu Blanca
```
Comment (JordiMForgeFlow) [inline stock_request_stage/__manifest__.py:1]:
```suggestion
# Copyright 2021 ForgeFlow, S.L.
```

**Comment by JordiMForgeFlow** [comment]
When trying to set a cover image in the kanban card:

image

**Comment by JordiMForgeFlow** [comment]
The button "Share" of the kanban is missing some method:

image

**Comment by AaronHForgeFlow** [comment]
Would it be possible to integrate this module with the base_kanban_stage module in server tools?

**Comment by JordiBForgeFlow** [comment]
@AaronHForgeFlow interesting idea! In this case, though, I am in need of some specific features like adding images or updating the states of the stock request. Also, it's been done already...

image

**Comment by rousseldenis** [comment]
@JordiBForgeFlow Could you rebase ?

**Comment by rousseldenis** [comment]
@JordiBForgeFlow

File: stock_request_stage/models/stock_request.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        order = "sequence"
+        return (
+            self.env["stock.request.stage"]
+            .search(search_domain, order=order, limit=1)
+            .id
+        )
+
+    stage_id = fields.Many2one(
+        "stock.request.stage",
+        default=_get_default_stage_id,
```
Comment (rousseldenis) [inline stock_request_stage/models/stock_request.py:22]: You can use lambda and put method under.

File: stock_request_stage/models/stock_request.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            elif set_state == "cancel" and rec.state != "cancel":
+                rec.action_cancel()
+            elif set_state == "done" and rec.state != "done":
+                rec.action_done()
+
+    def complete_pickings(self):
+        for rec in self:
+            if rec.stage_id.complete_pickings:
+                for picking in rec.picking_ids.filtered(lambda x: x.state == "draft"):
+                    picking.action_confirm()
```
Comment (rousseldenis) [inline stock_request_stage/models/stock_request.py:58]: Isn't it possible to use multi recordset for both actions ?

File: stock_request_stage/models/stock_request_stage.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+from odoo.addons.stock_request.models.stock_request import REQUEST_STATES
+
+
+class StockRequestStage(models.Model):
+    _name = "stock.request.stage"
+    _description = "Stock Request Stage"
+    _order = "sequence, id"
+
+    def _get_default_project_ids(self):
```
Comment (rousseldenis) [inline stock_request_stage/models/stock_request_stage.py:14]: Is this used somewhere ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Apart changes, @JordiBForgeFlow isn't this introducing complexity ?

Having two fields to represent a 'stage' is quite confusing for users.

However, could you be aware of 16 migrations as this can be merged in stock_request IMHO

**Pull request closed without merging**